src/abstractdataframe/abstractdataframe.jl

```diff
@@ -2060,10 +2060,23 @@ function Missings.disallowmissing(df::AbstractDataFrame,
     for i in axes(df, 2)
         x = df[!, i]
         if i in idxcols
-            if !error && Missing <: eltype(x) && any(ismissing, x)
-                y = x
+            if Missing <: eltype(x)
+                # finding missing is cheap so we accept paying it
+                # to get a better error message
+                row = findfirst(ismissing, x)
```

Comment:
Wouldn't it be more logical to do `try; y = disallowmissing(x); catch... ` and only then check whether `missing` is present? Is this check so cheap when there are no missing values in the data?

Reply:
I have thought about it. It adds around 20% to computation time:
```
julia> x = Union{Int, Missing}[1 for i in 1:10^8];

julia> @time findfirst(ismissing, x)
  0.063825 seconds (15.35 k allocations: 869.997 KiB, 23.10% compilation time)

julia> @time findfirst(ismissing, x)
  0.049497 seconds

julia> @time disallowmissing(x);
  0.210153 seconds (112.43 k allocations: 769.128 MiB, 1.87% gc time, 18.54% compilation time)

julia> @time disallowmissing(x);
  0.190172 seconds (2 allocations: 762.939 MiB, 7.27% gc time)

julia> @time disallowmissing(x);
  0.253140 seconds (2 allocations: 762.939 MiB, 32.77% gc time)
```
which I thought is acceptable (I do not assume that `disallowmissing` will be used in any hot loop - it is most likely a one time operation).

But if you prefer `try`-`catch` then I can change it as it would indeed save some time.

Reply:
OK - I have changed this.